### PR TITLE
[vili] Create a new port

### PR DESCRIPTION
**Pull request by nfarid**
**Describe the pull request**

- #### What does your PR fix?
  Adds the port for [Vili](https://github.com/ViliOrg/Vili).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All I presume, No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

### CODE REVIEW

**Comment by nfarid** [comment]
@microsoft-github-policy-service agree

**Comment by nfarid** [comment]
It seems that the static triplets work, but not the dynamic triplets.
I would need to fix this.

**Comment by nfarid** [comment]
I've made the changes, but before I make the PR for vili, I want to make sure that it works first.

**Comment by nfarid** [comment]
Great, it seems to work, I'm going to make PR for the Vili repo.

**Comment by nfarid** [comment]
@JonLiu1993 I believe this is ready to be merged.

File: ports/vili/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "vili",
+  "version": "1.0.0",
```
Comment (JonLiu1993) [inline ports/vili/vcpkg.json:3]: @nfarid, Thanks for your pr, the ref is not the v1.0.0 release commit, please use "1.0.0+20221123".

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/vili/vcpkg.json (hunk 1 above)
Comment (nfarid) [reply ports/vili/vcpkg.json]: @JonLiu1993 ref has been changed.

File: versions/v-/vili.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "versions": [
+    {
+      "git-tree": "b72194e61d1a2725cbb85912b25bc3e8b986e503",
+      "version": "1.0.0+20221123",
+      "port-version": 0
+    },
+    {
+      "git-tree": "cdd01422868aa5d5bdd735ae232cc1b49570b48c",
+      "version": "1.0.0",
+      "port-version": 0
```
Comment (JonLiu1993) [inline versions/v-/vili.json:11]:
```suggestion

```

Comment (JonLiu1993) [reply versions/v-/vili.json:11]: Each merge can only add one "git-tree".

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: versions/v-/vili.json (hunk 2 above)
Comment (nfarid) [reply versions/v-/vili.json]: @JonLiu1993  Changed and squash.

**Comment by JonLiu1993** [comment]
Successfully tested usage of the x64-windows triplet

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**